storage/buckets/buckets_test.go

```diff
@@ -31,42 +32,77 @@ import (
 	iampb "google.golang.org/genproto/googleapis/iam/v1"
 )
 
+const (
+	testPrefix      = "storage-buckets-test"
+	bucketExpiryAge = time.Hour * 24
+)
+
+func TestMain(m *testing.M) {
+	exit := m.Run()
+
+	tc, _ := testutil.ContextMain(m)
+	// Delete old buckets whose name begins with our test prefix
+	if err := testutil.DeleteExpiredBuckets(tc.ProjectID, testPrefix, bucketExpiryAge); err != nil {
+		// Don't fail the test if cleanup fails
+		log.Printf("Post-test cleanup failed: %v", err)
+	}
+	os.Exit(exit)
+}
+
 func TestCreate(t *testing.T) {
 	tc := testutil.SystemTest(t)
-	bucketName := tc.ProjectID + "-storage-buckets-tests"
+	bucketName := testutil.UniqueBucketName(testPrefix)
+	ctx := context.Background()
+
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		t.Fatalf("storage.NewClient: %v", err)
+	}
+	t.Cleanup(func() {
```

Comment:
`t.Cleanup` is new in Go 1.14-- do we have test runs for older versions in this repo? @codyoss

Reply:
We test on 1.11 and latest Go versions. In order to use this feature though you would need to add a go.mod file somewhere in storage directory tree that declares a Go version of at least 1.14. Our test runners trigger off of the language version specified in the go.mod.